### PR TITLE
chore: google analytics 2.0

### PR DESCRIPTION
**Pull request by macvogelsang**
Here's the index of all events emitted so far: https://docs.google.com/spreadsheets/d/1e_folq0RUvgL-2y96EsqjrpQAlSpGQTMjm0vG3SWYQM/edit#gid=0

One thing I couldn't figure out how to do was fetch node information when a node is clicked. I could only get the data when a node label was clicked. Please let me know how I can add this if it's desired.

I also decided not to track hover events, because that will lead to a ton of outgoing network requests.

### CODE REVIEW

File: projects/v2/src/app/models/ga.model.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Standard set of actions for GA events
+export enum GaAction {
+  // Action for form inputs
+  INPUT = 'input',
+  // Action for toggle ui elements
+  TOGGLE = 'toggle',
+  // Action for slider change enents
+  SLIDE = 'slide',
```
Comment (hrishikeshpaul) [inline projects/v2/src/app/models/ga.model.ts:8]: I haven't seen, but is this event fired everytime the slide is being used or is it fired when the slide action has been completed?

**Comment by hrishikeshpaul** [comment]
Great work! Try to remove as many code smells as possible!

File: projects/v2/src/app/models/ga.model.ts (hunk 1 above)
Comment (macvogelsang) [reply projects/v2/src/app/models/ga.model.ts:8]: They are only fired when the slide action is completed.

File: projects/v2/src/app/modules/tree/vega.service.ts [hunk 2]
```diff
@@ -71,21 +69,30 @@ export class VegaService {
...
-      };
-      this.ga.eventEmitter('graph_label_click', GaCategory.GRAPH, 'Clicked a node label', GaAction.CLICK, JSON.stringify(nodeInfo));
+      this.ga.eventEmitter('graph_label_click', GaCategory.GRAPH, 'Clicked a node label', GaAction.CLICK, this.makeNodeInfoString(node));
     });
 
     view.addSignalListener('node__click', (signal: Signal, nodeId: any) => {
-      // TODO get more node information
-      this.ga.eventEmitter('graph_node_click', GaCategory.GRAPH, 'Clicked a node', GaAction.CLICK, nodeId);
+      if (nodeId != null) {
+        console.log("clicked node " + nodeId);
```
Comment (hrishikeshpaul) [inline projects/v2/src/app/modules/tree/vega.service.ts:77]: Remove this log

File: projects/v2/src/app/modules/tree/vega.service.ts [hunk 3]
```diff
@@ -71,21 +69,30 @@ export class VegaService {
...
+        const node = this.store.selectSnapshot(TreeState.getBimodal).nodes.find(node => node.id === nodeId);
+        this.ga.eventEmitter('graph_node_select', GaCategory.GRAPH, 'Selected (clicked) a node', GaAction.CLICK, this.makeNodeInfoString(node));
+      } else {
+        this.ga.eventEmitter('graph_node_deselect', GaCategory.GRAPH, 'Deselected a node', GaAction.CLICK);
+      }
     });
   }
 
+  makeNodeInfoString(node: any) {
```
Comment (hrishikeshpaul) [inline projects/v2/src/app/modules/tree/vega.service.ts:86]: Can this be moved to the GA service or something? This tile should technically have only vega and visualization related files.

**Pull request merged**